cci_nepal/analysis/dummy_data/dummy_data_modeling/model_analysis.py

```diff
@@ -0,0 +1,295 @@
+import pandas as pd
+import numpy as np
+from numpy import absolute
+from numpy import mean
+from numpy import std
+from sklearn.model_selection import train_test_split
+from sklearn.metrics import accuracy_score
+from sklearn.linear_model import LinearRegression
+from sklearn.model_selection import cross_val_score
+from sklearn.model_selection import RepeatedKFold
+from sklearn.feature_selection import SequentialFeatureSelector
+from matplotlib import pyplot
+from operator import itemgetter
+
+import logging
+import cci_nepal
+from cci_nepal.getters.dummy_data import get_dummy_data as gdd
+
+
+def label_transformer(d):
+    if d < 1.5:
+        return "Unnecessary"
+    elif d < 2.5:
+        return "Desirable"
+    else:
+        return "Essential"
+
+
+def numeric_score_transformer(d):
+    if d < 1.5:
+        return 1
+    elif d < 2.5:
+        return 2
+    else:
+        return 3
+
+
+project_dir = cci_nepal.PROJECT_DIR
+logging.info(project_dir)
+
+
+df = gdd.read_csv_file(f"{project_dir}/inputs/data/Dummy_Data_Full_Survey_Features.csv")
+df.head()
+
+
+df.columns
+
+# Mapping categories to numbers #
+
+mapping = {"Essential": 3, "Desirable": 2, "Unnecessary": 1}
+df_dummy = df.applymap(lambda s: mapping.get(s) if s in mapping else s)
+
+
+df_dummy.head(5)
+
+# This part is just to test if all 3 labels are predicted well in the model ###
+
+# Since most of the predictions were otherwise averaged around 2 ###
+
+
+def replace_with_number(df, columns, values):
```

Comment:
Probably best to have all functions at the top. This will also help with refactoring as you can see more clearly what needs to be moved into pipeline / utils ect.

Again add doc-strings to the functions.